Library/Homebrew/formula_installer.rb

```diff
@@ -202,15 +202,38 @@ def verify_deps_exist
   def check_install_sanity
     raise FormulaInstallationAlreadyAttemptedError, formula if self.class.attempted.include?(formula)
 
+    deprecate_disable_reasons = {
+      does_not_build:      "does not build",
+      no_license:          "has no license",
+      repo_archived:       "has an archived upstream repository",
+      repo_removed:        "has a removed upstream repository",
+      unmaintained:        "is not maintained upstream",
+      unsupported:         "is not supported upstream",
+      deprecated_upstream: "is deprecated upstream",
+      versioned_formula:   "is a versioned formula",
+    }
+
     if formula.deprecated?
       if formula.deprecation_reason.present?
-        opoo "#{formula.full_name} has been deprecated because #{formula.deprecation_reason}!"
+        reason = if deprecate_disable_reasons.key? formula.deprecation_reason
+          deprecate_disable_reasons[formula.deprecation_reason]
+        else
+          deprecate_disable_reasons
+        end
+
+        opoo "#{formula.full_name} has been deprecated because it #{reason}!"
       else
         opoo "#{formula.full_name} has been deprecated!"
       end
     elsif formula.disabled?
       if formula.disable_reason.present?
-        odie "#{formula.full_name} has been disabled because #{formula.disable_reason}!"
+        reason = if deprecate_disable_reasons.key? formula.disable_reason
+          deprecate_disable_reasons[formula.disable_reason]
+        else
+          deprecate_disable_reasons
```

Comment:
`formula.disable_reason`?

Reply:
I think `formula.disable_reason` is a value, not a key. Same for `formula.deprecation_reason`

Reply:
I might be wrong but I thought that it's either a key when it's a symbol (e.g. `:does_not_build`) or a value when it's a string (e.g. `"is deprecated"`)

Reply:
Yeah it should be `formula.disable_reason`

Reply:
`deprecate_disable_reasons.key? formula.disable_reason` returns false for `semtag`

Reply:
The line in `semtag` currently is:

```ruby
disable! because: "has no license"
```

so `formula.disable_reason` returns `"has no license"`. That's not a key in `deprecate_disable_reasons` so it will default to using whatever text is in `formula.disable_reason` as the message.

Now that this PR is merged and tagged, the line in `semtag` can become:

```ruby
disable! because: :no_license
```

Now, `formula.disable_reason` returns `:no_license` which _is_ a key in `deprecate_disable_reasons` so the message will use the value of `"has no license"`.

Reply:
Thanks for the clarification, Rylan! This makes perfect sense now.